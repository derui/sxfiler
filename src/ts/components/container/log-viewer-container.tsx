import { h } from "preact";
import { Component as LogViewer } from "@/components/project/log-viewer";
import { State } from "@/modules";

export type Props = {
  state: State;
};

export const Component: preact.FunctionComponent<Props> = ({ state }) => {
  return (
    <div class="log-viewer-container__root" data-testid="logViewerContainer">
      <LogViewer entries={state.logEvent.events} hidden={false} />
    </div>
  );
};
