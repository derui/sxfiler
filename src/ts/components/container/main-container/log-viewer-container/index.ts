export * from "./log-viewer-container";
