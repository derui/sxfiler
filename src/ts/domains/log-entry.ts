export enum Level {
  Info = "info",
  Warning = "warning",
  Error = "error",
}

export type LogEntryObject = {
  id: string;
  level: Level;
  body: string;
};

export type LogEntry = LogEntryObject;

/**
 * create new log entry from arguments
 */
export const createLogEntry = ({ id, level, body }: { id: string; level: Level; body: string }): LogEntry => {
  return {
    id,
    level,
    body,
  };
};
