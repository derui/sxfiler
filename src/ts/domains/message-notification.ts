export enum Level {
  Info = "info",
  Warning = "warning",
  Error = "error",
}

export type MessageNotification = {
  readonly id: string;
  readonly level: Level;
  readonly body: string;
};

/**
 * factory function to create one-shot notification
 * @param id ID of a notification
 * @param level level of a notification
 * @param message body of a message notification
 */
export const createMessage = function createMessage({
  id,
  level,
  message,
}: {
  id: string;
  level: Level;
  message: string;
}): MessageNotification {
  return { id, level, body: message };
};
