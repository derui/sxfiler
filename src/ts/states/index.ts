import * as Config from "./config";
import * as FileList from "./file-list";
import * as Keymap from "./keymap";
import * as Notification from "./notification";
import * as TaskInteraction from "./task-interaction";
import * as LogEntry from "./log-entry";
import * as H from "./history";
import * as F from "./finder";
import * as Completer from "./completer";
import { AppContext, createAppContext } from "@/domains/app-context";
import { UIContext } from "@/types/ui-context";

export type AppState = {
  context: AppContext;
  config: Config.State;
  fileList: FileList.State;
  keymap: Keymap.State;
  notification: Notification.State;
  taskInteraction: TaskInteraction.State;
  logEntry: LogEntry.State;
  history: H.State;
  finder: F.State;
  completer: Completer.State;
};

// get empty state
export const empty = function emptyAppState(): AppState {
  return {
    context: createAppContext({ current: UIContext.OnFileTree }),
    config: Config.empty(),
    fileList: FileList.empty(),
    keymap: Keymap.empty(),
    notification: Notification.empty(),
    taskInteraction: TaskInteraction.empty(),
    logEntry: LogEntry.empty(),
    history: H.empty(),
    finder: F.empty(),
    completer: Completer.empty(),
  };
};
