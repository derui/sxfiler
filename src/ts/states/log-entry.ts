import { MessageNotification } from "../domains/message-notification";

// define state to store log entries

export type State = {
  entries: Map<string, MessageNotification>;
};

// get new state
export const empty = (): State => {
  return {
    entries: new Map(),
  };
};

// push new entry to state
export const pushEntry = (state: State, obj: MessageNotification): State => {
  const newEntry = new Map(state.entries);

  newEntry.set(obj.id, obj);
  return {
    entries: newEntry,
  };
};
