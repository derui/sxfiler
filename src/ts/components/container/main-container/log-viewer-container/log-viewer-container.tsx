import * as React from "react";
import { styled } from "@/components/theme";

import * as Element from "@/components/ui/element";
import { State } from "@/states/log-entry";
import { Component as LogViewer } from "@/components/project/log-viewer";

export type Props = {
  state: State;
};

const Root = styled(Element.Component)`
  position: relative;

  height: 100%;
  width: 100%;
  max-width: 100%;
  border-top: 1px solid ${props => props.theme.colors.base2};
`;

export const Component: React.FC<Props> = ({ state }) => {
  return (
    <Root>
      <LogViewer entries={Array.from(state.entries.values())} hidden={false} />
    </Root>
  );
};
