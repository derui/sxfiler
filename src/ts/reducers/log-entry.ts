// reducers for notification
import { Actions } from "../actions";
import { ActionTypes } from "../actions/notification";
import { empty, State } from "../states/log-entry";

export function reducer(state: State = empty(), action: Actions): State {
  switch (action.type) {
    case ActionTypes.receiveMessage:
      let v = new Map(state.entries);
      v.set(action.notification.id, action.notification);
      return { ...state, entries: v };
  }
  return state;
}

export default reducer;
