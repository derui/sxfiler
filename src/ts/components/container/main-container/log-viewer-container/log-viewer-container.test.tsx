import * as React from "react";
import renderer from "react-test-renderer";

import { Component } from "./log-viewer-container";
import { empty, pushEntry } from "@/states/log-entry";
import { Level, createMessage } from "@/domains/message-notification";

describe("Container", () => {
  describe("Log Viewer Container", () => {
    it("should render correctly", () => {
      const state = pushEntry(empty(), createMessage({ id: "id", level: Level.Info, message: "message" }));
      const tree = renderer.create(<Component state={state} />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
