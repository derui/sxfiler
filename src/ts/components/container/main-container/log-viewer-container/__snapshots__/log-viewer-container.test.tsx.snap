// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Log Viewer Container should render correctly 1`] = `
<div
  className="sc-ifAKCX fskjUa"
>
  <div
    aria-hidden={false}
    className="sc-bdVaJa kMdzPL"
    role="list"
  >
    <div
      aria-selected={false}
      className="sc-bwzfXH daSFCf"
      role="listitem"
    >
      <span
        className="sc-htpNat kQdIEM"
        data-level="info"
      >
        [
        INFO
        ]
      </span>
      <span
        className="sc-bxivhb ghzfYa"
      >
        message
      </span>
    </div>
  </div>
</div>
`;
